source/representing-replays.rst

```diff
@@ -0,0 +1,144 @@
+Representing Replays
+====================
+
+Circlecore needs a replay to investigate before it can tell you anything
+about if that replay is cheated. There are several ways to create a replay
+through circlecore.
+
+All four of the following classes are subclasses of |Loadable|.
+
+.. note::
+
+    All of the following objects are lazy-loaded. That is, they are as cheap
+    as possible to instantiate, and only incur a loading penalty when
+    necessary (see |cg.load|, |cg.load_info|, and |cg.run|).
+
+
+Replay
+------
+
+A |Replay| is the most basic object of circlecore. One replay object represents
+one play made by a user. We define two |Replay| subclasses.
+
+ReplayMap
+~~~~~~~~~
+
+|ReplayMap| represents a replay by a user on a map. For instance,
+
+.. code-block:: python
+
+    r1 = ReplayMap(221777, 2757689)
+
+represents the highest scoring replay by user ``2757689`` on map ``221777``. To
+restrict this to a certain replay, specify a mods argument. This represents
+specifically the ``HDHR`` play by the same user on the same map.
+
+.. code-block:: python
+
+    r1 = ReplayMap(221777, 2757689, mods=Mod.HD + Mod.HR)
+
+While in this case these are identical replays, there are times a user has two
+available replays on a map, with different mods. Because of how osu! stores
+replays, it is impossible to have two available replays with the same
+mod combination.
+
+ReplayPath
+~~~~~~~~~~
+
+|ReplayPath| represents a replay stored locally in an ``osr`` file. For
+instance,
+
+.. code-block:: python
+
+    r2 = ReplayPath("/Users/tybug/Desktop/replays/replay1.osr")
```

Comment:
```suggestion
    r2 = ReplayPath("./tests/resources/example_replay.osr")
```

Reply:
This code isn't meant to be run in a circlecore directory; I'm just giving an example path. Maybe `"/path/to/your/replay.osr` is better, but I think a real example of a path works too.

Using the test osrs is just confusing.

Reply:
yea, `"/path/to/your/replay.osr"` works way better, `"/Users/tybug/Desktop/replays/replay1.osr"` just feels unpolished imo.